include/onnxruntime/core/framework/tensor.h

```diff
@@ -189,7 +189,7 @@ class Tensor final {
     ORT_ENFORCE(utils::IsPrimitiveDataType<T>(dtype_), "Tensor type mismatch. ",
                 "T ", "!=", dtype_);
     const T* data = reinterpret_cast<const T*>(static_cast<char*>(p_data_) + byte_offset_);
-    return gsl::make_span(data, static_cast<typename gsl::span<T>::index_type>(shape_.Size()));
+    return gsl::make_span(data, static_cast<typename gsl::span<T>::size_type>(shape_.Size()));
```

Comment:
MS GSL's span has a size_type which is size_t. it doesn't have an index_type.
https://github.com/microsoft/GSL/blob/10df83d292bf5bbdc487e57dc8c2dc8c7a01f4d1/include/gsl/span#L426